### PR TITLE
Add quarkus.servlet.max-parameters setting

### LINKED ISSUE #63: Ability to set MAX_PARAMETERS in Undertow
When using Undertow, a request with more than 1,000 parameters triggers a `io.undertow.util.ParameterLimitException`.  While that's a reasonable default, I have a need to configure it.  (A JSF application I am porting to Quarkus via the MyFaces Quarkus extension generates very large requests.)

Undertow provides a way of providing it with an `UndertowOptionMap`, one key of which is `MAX_PARAMETERS`.  It would be nice if there were a way of setting it through `application.properties`.

### PR DESCRIPTION
**Pull request by benkard**
This configures the MAX_PARAMETERS setting in Undertow, which defaults
to 1,000.

Makes use of the UndertowOptionsMap setter introduced in quarkus-http
commit 00223f28f17a246a3075b3f1ce38dc289976a641 and therefore depends on
quarkus-http 3.1.0.Beta2.

Fixes quarkusio/quarkus-http#63.

### CODE REVIEW

**Comment by benkard** [comment]
(The CI pipeline is expected to fail until quarkus-http 3.1.0.Beta2 is released.)

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletRuntimeConfig.java
Comment (stuartwdouglas) [inline extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletRuntimeConfig.java:33]: I don't know if it makes sense to link to the Undertow 2.x docs here, as Quarkus HTTP is not the same thing.

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java
Comment (stuartwdouglas) [inline extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java:399]: This map should be built once, not per-request.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java
Comment (benkard) [reply extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java]: Ah, right.  Done.

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletRuntimeConfig.java
Comment (benkard) [reply extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletRuntimeConfig.java]: Done – replaced it with a sentence describing the effect.

**Comment by benkard** [comment]
A thought: Should the option be under a separate `quarkus.servlet.limits` group, similarly to how there is `quarkus.http.limits`?

**Review by gsmet — DISMISSED** [review]
This will need a squash before merging. Thanks!

I'll let @stuartwdouglas approve.

**Comment by benkard** [comment]
> This will need a squash before merging. Thanks!

Done.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by benkard** [comment]
@gsmet I think your change request is still blocking the merge.  Could you have a look, please?  Thanks!

**Comment by gsmet** [comment]
I rebased and pushed a small additional commit to remove the use of Optional as we have a default value. Let's wait for CI and merge!

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**
